File: SearchSVC/clsFILESTODELETE.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql


Public Class clsFILESTODELETE


    '** DIM the selected table columns 
    Dim DB As New clsDatabaseSVR
    Dim DMA As New clsDmaSVR
    Dim UTIL As New clsUtilitySVR

    Dim DG As New clsDataGrid(SecureID)


    Dim UserID As String = ""
    Dim MachineName As String = ""
    Dim FQN As String = ""
    Dim PendingDelete As String = ""

       Dim SecureID As Integer = -1

    Sub New(ByVal iSecureID As Integer)
        SecureID = iSecureID
    End Sub


    '** Generate the SET methods 
    ''' <summary>
    ''' Sets the userid.
    ''' </summary>
    ''' <param name="val">The value.</param>
    Public Sub setUserid(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        UserID = val
    End Sub


    Public Sub setMachinename(ByRef val as string)
        val = UTIL.RemoveSingleQuotes(val)
        MachineName = val
    End Sub


    Public Sub setFqn(ByRef val as string)
        val = UTIL.RemoveSingleQuotes(val)
        FQN = val
    End Sub


    Public Sub setPendingdelete(ByRef val as string)
        If Len(val) = 0 Then
            val = "null"
        End If
        If val.Equals("Y") Then val = "1"
        If val.Equals("N") Then val = "0"
        val = UTIL.RemoveSingleQuotes(val)
        PendingDelete = val
    End Sub






    '** Generate the GET methods 
    Public Function getUserid() As String
        Return UTIL.RemoveSingleQuotes(UserID)
    End Function


    Public Function getMachinename() As String
        Return UTIL.RemoveSingleQuotes(MachineName)
    End Function


    Public Function getFqn() As String
        Return UTIL.RemoveSingleQuotes(FQN)
    End Function


    Public Function getPendingdelete() As String
        If Len(PendingDelete) = 0 Then
            PendingDelete = "null"
        End If
        Return PendingDelete
    End Function






    '** Generate the Required Fields Validation AS string 
    Public Function ValidateReqData() As Boolean
        Return True
    End Function




    '** Generate the Validation AS string 
    Public Function ValidateData() As Boolean
        Return True
    End Function

    '** Generate the INSERT AS string 
    Public Function Insert() As Boolean
        Dim b As Boolean = false
        Dim s As String = ""
        s = s + " INSERT INTO FilesToDelete("
        s = s + "UserID,"
        s = s + "MachineName,"
        s = s + "FQN,"
        s = s + "PendingDelete) values ("
        s = s + "'" + UserID + "'" + ","
        s = s + "'" + MachineName + "'" + ","
        s = s + "'" + FQN + "'" + ","
        s = s + PendingDelete + ")"
        Return DB.DBExecuteSql(SecureID, S, False)


    End Function




    '** Generate the UPDATE AS string 
    Public Function Update(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        s = s + " update FilesToDelete set "
        s = s + "UserID = '" + getUserid() + "'" + ", "
        s = s + "MachineName = '" + getMachinename() + "'" + ", "
        s = s + "FQN = '" + getFqn() + "'" + ", "
        s = s + "PendingDelete = " + getPendingdelete()
        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DB.DBExecuteSql(SecureID, S, False)
    End Function




    '** Generate the SELECT AS string 
    Public Function SelectRecs() As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "UserID,"
        s = s + "MachineName,"
        s = s + "FQN,"
        s = s + "PendingDelete "
        s = s + " FROM FilesToDelete"

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the Select One Row AS string 
    Public Function SelectOne(ByVal WhereClause as string) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "UserID,"
        s = s + "MachineName,"
        s = s + "FQN,"
        s = s + "PendingDelete "
        s = s + " FROM FilesToDelete"
        s = s + WhereClause

        Dim CS$ = DBgetConnStr() : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function




    '** Generate the DELETE AS string 
    Public Function Delete(ByVal WhereClause as string) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        If Len(WhereClause) = 0 Then Return False


        WhereClause = " " + WhereClause


        s = " Delete from FilesToDelete"
        s = s + WhereClause


        b = DB.DBExecuteSql(SecureID, S, False)
        Return b


    End Function




    '** Generate the Zeroize Table AS string 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""


        s = s + " Delete from FilesToDelete"


        b = DB.DBExecuteSql(SecureID, S, False)
        Return b


    End Function

    '** Generate Index Queries 
    Public Function cnt_PK_FileToDelete(ByVal SecureID As Integer, ByVal FQN As String, ByVal MachineName As String, ByVal UserID As String) As Integer
        Dim B As Integer = 0
        Dim TBL$ = "FilesToDelete"
        Dim WC$ = "Where FQN = '" + FQN + "' and   MachineName = '" + MachineName + "' and   UserID = '" + UserID + "'"
        B = DB.iGetRowCount(TBL$, WC)
        Return B
    End Function     '** cnt_PK_FileToDelete


    '** Generate Index ROW Queries 
    Public Function getRow_PK_FileToDelete(ByVal FQN As String, ByVal MachineName As String, ByVal UserID As String) As SqlDataReader


        Dim rsData As SqlDataReader = Nothing
        Dim TBL$ = "FilesToDelete"
        Dim WC$ = "Where FQN = '" + FQN + "' and   MachineName = '" + MachineName + "' and   UserID = '" + UserID + "'"


        rsData = DB.GetRowByKey(SecureID, TBL$, WC)


        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PK_FileToDelete


    ''' Build Index Where Caluses 
    '''
    Public Function wc_PK_FileToDelete(ByVal FQN As String, ByVal MachineName As String, ByVal UserID As String) As String


        Dim WC$ = "Where FQN = '" + FQN + "' and   MachineName = '" + MachineName + "' and   UserID = '" + UserID + "'"


        Return WC
    End Function     '** wc_PK_FileToDelete


    '** Generate the SET methods 


End Class

File: Archiver/CS/clsFILESTODELETE.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Drawing;
using System.Linq;
using System.Diagnostics;
using System.Data;
using Microsoft.VisualBasic;
using MODI;
using System.Xml.Linq;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.Configuration;
using System.Data.SqlClient;
using System.IO;
using System.Data.Sql;



namespace EcmArchiveClcSetup
{
	public class clsFILESTODELETE
	{
		
		
		//** DIM the selected table columns
		clsDatabase DB = new clsDatabase();
		clsDma DMA = new clsDma();
		clsUtility UTIL = new clsUtility();
		
		clsDataGrid DG = new clsDataGrid();
		
		
		string UserID = "";
		string MachineName = "";
		string FQN = "";
		string PendingDelete = "";
		
		
		
		
		//** Generate the SET methods
		public void setUserid(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			UserID = val;
		}
		
		
		public void setMachinename(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			MachineName = val;
		}
		
		
		public void setFqn(ref string val)
		{
			val = UTIL.RemoveSingleQuotes(val);
			FQN = val;
		}
		
		
		public void setPendingdelete(ref string val)
		{
			if (val.Length == 0)
			{
				val = "null";
			}
			if (val.Equals("Y"))
			{
				val = "1";
			}
			if (val.Equals("N"))
			{
				val = "0";
			}
			val = UTIL.RemoveSingleQuotes(val);
			PendingDelete = val;
		}
		
		
		
		
		
		
		//** Generate the GET methods
		public string getUserid()
		{
			return UTIL.RemoveSingleQuotes(UserID);
		}
		
		
		public string getMachinename()
		{
			return UTIL.RemoveSingleQuotes(MachineName);
		}
		
		
		public string getFqn()
		{
			return UTIL.RemoveSingleQuotes(FQN);
		}
		
		
		public string getPendingdelete()
		{
			if (PendingDelete.Length == 0)
			{
				PendingDelete = "null";
			}
			return PendingDelete;
		}
		
		
		
		
		
		
		//** Generate the Required Fields Validation method
		public bool ValidateReqData()
		{
			return true;
		}
		
		
		
		
		//** Generate the Validation method
		public bool ValidateData()
		{
			return true;
		}
		
		//** Generate the INSERT method
		public bool Insert()
		{
			bool b = false;
			string s = "";
			s = s + " INSERT INTO FilesToDelete(";
			s = s + "UserID,";
			s = s + "MachineName,";
			s = s + "FQN,";
			s = s + "PendingDelete) values (";
			s = s + "\'" + UserID + "\'" + ",";
			s = s + "\'" + MachineName + "\'" + ",";
			s = s + "\'" + FQN + "\'" + ",";
			s = s + PendingDelete + ")";
			return DB.ExecuteSqlNewConn(s, false);
			
			
		}
		
		
		
		
		//** Generate the UPDATE method
		public bool Update(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			
			s = s + " update FilesToDelete set ";
			s = s + "UserID = \'" + getUserid() + "\'" + ", ";
			s = s + "MachineName = \'" + getMachinename() + "\'" + ", ";
			s = s + "FQN = \'" + getFqn() + "\'" + ", ";
			s = s + "PendingDelete = " + getPendingdelete();
			WhereClause = (string) (" " + WhereClause);
			s = s + WhereClause;
			return DB.ExecuteSqlNewConn(s, false);
		}
		
		
		
		
		//** Generate the SELECT method
		public SqlDataReader SelectRecs()
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "UserID,";
			s = s + "MachineName,";
			s = s + "FQN,";
			s = s + "PendingDelete ";
			s = s + " FROM FilesToDelete";
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		
		
		//** Generate the Select One Row method
		public SqlDataReader SelectOne(string WhereClause)
		{
			bool b = false;
			string s = "";
			SqlDataReader rsData;
			s = s + " SELECT ";
			s = s + "UserID,";
			s = s + "MachineName,";
			s = s + "FQN,";
			s = s + "PendingDelete ";
			s = s + " FROM FilesToDelete";
			s = s + WhereClause;
			
			string CS = DB.getGateWayConnStr(modGlobals.gGateWayID);
			SqlConnection CONN = new SqlConnection(CS);
			CONN.Open();
			SqlCommand command = new SqlCommand(s, CONN);
			rsData = command.ExecuteReader();
			return rsData;
		}
		
		
		
		
		//** Generate the DELETE method
		public bool Delete(string WhereClause)
		{
			bool b = false;
			string s = "";
			
			
			if (WhereClause.Length == 0)
			{
				return false;
			}
			
			
			WhereClause = (string) (" " + WhereClause);
			
			
			s = " Delete from FilesToDelete";
			s = s + WhereClause;
			
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
			
		}
		
		
		
		
		//** Generate the Zeroize Table method
		public bool Zeroize()
		{
			bool b = false;
			string s = "";
			
			
			s = s + " Delete from FilesToDelete";
			
			
			b = DB.ExecuteSqlNewConn(s, false);
			return b;
			
			
		}
		
		
		
		
		//** Generate Index Queries
		public int cnt_PK_FileToDelete(string FQN, string MachineName, string UserID)
		{
			
			
			int B = 0;
			string TBL = "FilesToDelete";
			string WC = "Where FQN = \'" + FQN + "\' and   MachineName = \'" + MachineName + "\' and   UserID = \'" + UserID + "\'";
			
			
			B = DB.iGetRowCount(TBL, WC);
			
			
			return B;
		} //** cnt_PK_FileToDelete
		
		
		//** Generate Index ROW Queries
		public SqlDataReader getRow_PK_FileToDelete(string FQN, string MachineName, string UserID)
		{
			
			
			SqlDataReader rsData = null;
			string TBL = "FilesToDelete";
			string WC = "Where FQN = \'" + FQN + "\' and   MachineName = \'" + MachineName + "\' and   UserID = \'" + UserID + "\'";
			
			
			rsData = DB.GetRowByKey(TBL, WC);
			
			
			if (rsData.HasRows)
			{
				return rsData;
			}
			else
			{
				return null;
			}
		} //** getRow_PK_FileToDelete
		
		
		/// Build Index Where Caluses
		///
		public string wc_PK_FileToDelete(string FQN, string MachineName, string UserID)
		{
			
			
			string WC = "Where FQN = \'" + FQN + "\' and   MachineName = \'" + MachineName + "\' and   UserID = \'" + UserID + "\'";
			
			
			return WC;
		} //** wc_PK_FileToDelete
		
		
		//** Generate the SET methods
		
		
	}
	
}
